src/components/ApiAction.vue

```diff
@@ -0,0 +1,416 @@
+      this.savedQueries.push(tab)
+      this.newSaveTabIdx = null
+      this.storeQueriesToLocalStorage()
+    },
+    saveQuery(tabIdx) {
+      const tab = this.tabs[tabIdx]
+      const storedQueryIdx = this.savedQueries.findIndex(
+        q => q.name === tab.name
+      )
+      if (storedQueryIdx !== -1) {
+        this.savedQueries[storedQueryIdx].query = JSON.parse(
+          JSON.stringify(tab.query)
+        )
+        this.tabs[tabIdx].saved = true
+        this.storeQueriesToLocalStorage()
+      } else {
+        this.newSaveTabIdx = tabIdx
+        this.$bvModal.show('modal-save-query')
+      }
+    },
+    loadStoredQueriesFromLocalStorage() {
+      const storedQueries = localStorage.getItem('storedQueries')
+      if (!storedQueries) {
+        return
+      }
+      const jsonStoreQueries = JSON.parse(storedQueries)
+      if (!jsonStoreQueries[this.currentEnvironment.name]) {
+        return
+      }
+      this.savedQueries = jsonStoreQueries[this.currentEnvironment.name].map(
+        (q, idx) => {
+          q.response = ''
+          q.saved = true
+          q.savedIdx = idx
+          return q
+        }
+      )
+    },
+    storeQueriesToLocalStorage() {
+      let storedQueries = JSON.parse(localStorage.getItem('storedQueries'))
+      if (!storedQueries) {
+        storedQueries = {}
+      }
+      const queriesToStore = JSON.parse(JSON.stringify(this.savedQueries))
+      storedQueries[this.currentEnvironment.name] = queriesToStore.map(q => {
+        delete q.response
+        delete q.saved
+        delete q.savedIdx
+        return q
+      })
+      localStorage.setItem('storedQueries', JSON.stringify(storedQueries))
+    },
+    async performQuery(tabIdx) {
+      const query = JSON.parse(JSON.stringify(this.tabs[tabIdx].query))
+      let response = {}
+
+      try {
+        response = await this.$kuzzle.query(query)
+      } catch (error) {
+        response = {
+          ...error,
+          message: error.message,
+          stack: error.stack,
+          kuzzleStack: error.kuzzleStack
+        }
+      }
+      this.tabs[tabIdx].response = response
+    },
+    newTab() {
+      let newTab = JSON.parse(JSON.stringify(this.emptyTab))
+      this.tabs.push(newTab)
+      this.currentTabIdx = this.tabs.length - 1
+    },
+    async getKuzzleOpenApi() {
+      try {
+        const openApi = await this.$kuzzle.query({
+          controller: 'server',
+          action: 'openapi'
+        })
+        this.openapi = openApi.paths
+      } catch (error) {
+        this.$log.error(error)
+      }
+    },
+    async getKuzzlePublicApi() {
+      try {
+        const publicApi = await this.$kuzzle.query({
+          controller: 'server',
+          action: 'publicApi'
+        })
+        this.api = publicApi.result
+      } catch (error) {
+        this.$log.error(error)
+      }
+    }
+  },
+  async mounted() {
+    this.loading = true
+    if (this.canGetPublicApi) {
```

Comment:
And what if he can't? Should show some feedback

Reply:
The openapi and publicapi actions are called only to fill the datalist on controllers/actions inputs
If the user have no rights about that two actions, the inputs still usable, they just dont have the datalist options, so I'm not sure the user should be noticed about that :/

Reply:
I'd add some kind of "Unable to fetch the API action list" as  a disabled option in the drop down.